src/app/beer-detail/beer-detail.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Location } from '@angular/common';
+import { Beer } from '../beer';
+import { BeerService } from '../beer.service';
+
+@Component({
+    templateUrl: './beer-detail.component.html',
+    styleUrls: ['./beer-detail.component.css']
+})
+export class BeerDetailComponent implements OnInit {
+    @Input() beer: Beer;
+
+    constructor(
+        private route: ActivatedRoute,
+        private beerService: BeerService,
+        private location: Location
+    ) { }
+
+    ngOnInit() {
+        this.getBeer();
+    }
+    private getBeer(): void {
+        const id = +this.route.snapshot.paramMap.get('id');
+        this.beerService.getBeer(id)
+            .subscribe(beer => this.beer = beer);
```

Comment:
^instead: 
 this.route.paramMap.pipe(
       switchMap((params) => {
            const beerId = params.get('id');
                return this.beerService.getBeer(beerId);
            })
        )
        .subscribe((beer: Beer) => {

Reply:
this will react to the change in route parameters, and with every new id in parameters, it will re-load the beer.